src/cli/net/server/handlers/mul.rs

```diff
@@ -0,0 +1,106 @@
+use crate::cli::net::server::MpcServerError;
+use crate::cli::net::{BufferedMessages, RecordHeaders};
+use crate::protocol::{QueryId, Step};
+use async_trait::async_trait;
+use axum::body::Bytes;
+use axum::extract::{self, FromRequest, RequestParts};
+use tokio::sync::mpsc;
+
+pub struct Path<S: Step>(QueryId, S);
+#[async_trait]
+impl<B: Send, S: Step> FromRequest<B> for Path<S> {
+    type Rejection = MpcServerError;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        let extract::Path((query_id, step)) =
+            extract::Path::<(QueryId, String)>::from_request(req).await?;
+        let step = S::try_from(step).map_err::<serde_json::Error, _>(serde::de::Error::custom)?;
+        Ok(Path(query_id, step))
+    }
+}
+
+/// accepts all the relevant information from the request, and push all of it onto the `outgoing`
+/// channel
+/// TODO: implement the receiving end of `outgoing`
+pub async fn handler<S: Step>(
+    outgoing: mpsc::Sender<BufferedMessages<S>>,
```

Comment:
I think we should have a stateful router that injects the entry point to networking layer through the extensions mechanism. Injecting individual channels like this works too, but it can quickly grow unmaintainable as we would be managing individual channels for every handler (or share a single channel between some of them and use unique channels for others).

I agree that handlers should do almost no work and after parsing and validating the request, pass the payload to the networking layer, but I think we should make a "gateway" to it available to handlers through extensions

Reply:
btw converting bytes to a vector of messages is probably a good use case for `spawn_blocking` because it may block scheduler for longer that we would want.

Reply:
i'll keep `spawn_blocking` in mind when I get to that part of the implementation.

i also agree with your gateway idea. I think, for the purposes of this PR, it might make sense to just leave this as a channel here, with the idea that i'll swap it out for a gateway, for effectively the same functionality. However, I don't have a problem switching to an `Extensions` implementation. that's an interesting idea, actually, because it could parse the `query_id` and `step`, then inject the correct gateway from a `HashMap` of all existing gateways

Reply:
obsolete code